cmd/channel/email/main.go

```diff
@@ -38,17 +46,50 @@ func (ch *Email) SendNotification(req *plugin.NotificationRequest) error {
 	}
 
 	var msg bytes.Buffer
-	_, _ = fmt.Fprintf(&msg, "To: %s\n", strings.Join(to, ","))
-	_, _ = fmt.Fprintf(&msg, "Subject: [#%d] %s %s is %s\n\n", req.Incident.Id, req.Event.Type, req.Object.Name, req.Incident.Severity)
-
 	plugin.FormatMessage(&msg, req)
 
-	err := smtp.SendMail(ch.GetServer(), nil, ch.From, to, bytes.ReplaceAll(msg.Bytes(), []byte("\n"), []byte("\r\n")))
+	return enmime.Builder().
+		ToAddrs(to).
+		From(ch.SenderName, ch.SenderMail).
+		Subject(fmt.Sprintf("[#%d] %s %s is %s", req.Incident.Id, req.Event.Type, req.Object.Name, req.Incident.Severity)).
+		Header("Message-Id", fmt.Sprintf("<%d-%x-%s>", req.Incident.Id, time.Now().UnixNano(), ch.SenderMail)).
+		Text(msg.Bytes()).
+		Send(ch)
+}
+
+func (ch *Email) Send(reversePath string, recipients []string, msg []byte) error {
+	var (
+		client *smtp.Client
+		err    error
+	)
+
+	if ch.Encryption == "tls" {
+		client, err = smtp.DialTLS(ch.GetServer(), nil)
+	} else {
+		client, err = smtp.Dial(ch.GetServer())
+	}
 	if err != nil {
 		return err
 	}
+	defer client.Close()
 
-	return nil
+	if ch.Encryption == "starttls" {
+		if err = client.StartTLS(nil); err != nil {
+			return err
+		}
+	}
```

Comment:
What's the default certificate validation behavior of `DialTLS()` and `StartTLS()`?

Reply:
The `serverName` is required to validate the certificate. We already pass this to `smtp.DialTLS/smtp.Dial`.
It is then passed on and during the handshake it is checked whether the certificates are valid.